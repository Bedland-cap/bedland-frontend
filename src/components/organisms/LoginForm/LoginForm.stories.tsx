import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import LoginForm from './LoginForm';

const meta = {
  title: 'organisms/LoginForm',
  component: LoginForm,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        <BrowserRouter>{Story()}</BrowserRouter>
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof LoginForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
