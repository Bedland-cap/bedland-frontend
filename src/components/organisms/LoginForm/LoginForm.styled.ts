import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { PaletteType } from 'utils/types/theme.types';

const LoginFormContainer = styled.form`
  align-items: center;
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  max-width: 25.625rem;
  padding: 0rem 4rem 1.5rem 4rem;
`;

export const ResetPasswordLink = styled(Link)<{ palette: PaletteType }>`
  color: ${({ palette }) => (palette as PaletteType).secondaryDark};
  display: flex;
  justify-content: center;
`;

export default LoginFormContainer;
