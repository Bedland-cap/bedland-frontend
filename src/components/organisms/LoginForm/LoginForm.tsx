import { useForm, SubmitHandler, FieldValues } from 'react-hook-form';
import { useContext } from 'react';
import ErrorMessage from 'components/atoms/ErrorMessage/ErrorMessage';
import { useNavigate, Link } from 'react-router-dom';
import InputWithLabel from 'components/molecules/InputWithLabel/InputWithLabel';
import Button from 'components/atoms/Button/Button';
import { ThemeContext } from 'theme/ThemeContext';
import AlertIcon from 'assets/AlertIcon.svg';
import Typography from 'components/atoms/Typography/Typography';
import { useAppDispatch } from 'store/hooks';
import { login, User } from 'store/reducers/user_slice';
import LoginFormContainer from './LoginForm.styled';
import loginRegexPattern from './loginRegexPatterns';

const LoginForm = () => {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const { palette } = useContext(ThemeContext);
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  const onSubmit: SubmitHandler<FieldValues> = (data: FieldValues) => {
    const testUser: Omit<User, 'loggedIn'> = {
      userId: '122',
      login: data.login,
      status: 'success',
    };
    dispatch(login(testUser));
    navigate('/');
  };

  return (
    <LoginFormContainer onSubmit={handleSubmit(onSubmit)}>
      <Typography variant='sectionTitle' color='light' style={{ textAlign: 'center' }}>
        Login
      </Typography>
      <InputWithLabel
        label='Login'
        input='login'
        placeholder='Enter your login'
        type='text'
        register={register}
        regexPattern={loginRegexPattern.login}
      />
      <div>
        <InputWithLabel
          label='Password'
          input='password'
          placeholder='Enter your password'
          type='password'
          register={register}
          regexPattern={loginRegexPattern.password}
        />

        {errors.password || errors.login ? (
          <ErrorMessage>
            <img src={AlertIcon} alt='' />
            <div>
              Login or password are incorrect, please try again or&nbsp;
              <Link to='/#' style={{ color: 'red' }}>
                reset your password
              </Link>
            </div>
          </ErrorMessage>
        ) : null}
      </div>
      <Button
        backgroundColor={palette.secondaryDark}
        textColor='#ffffff'
        type='submit'
        variant='primary'
      >
        Login
      </Button>
    </LoginFormContainer>
  );
};

export default LoginForm;
