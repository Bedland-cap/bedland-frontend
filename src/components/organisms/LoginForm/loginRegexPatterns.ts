import { LoginRegexProps } from 'components/atoms/Input/Input';

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^[a-zA-Z]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
