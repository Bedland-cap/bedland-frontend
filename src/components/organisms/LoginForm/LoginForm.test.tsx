import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import LoginForm from './LoginForm';

describe('Login Form', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <BrowserRouter>
            <LoginForm />
          </BrowserRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
