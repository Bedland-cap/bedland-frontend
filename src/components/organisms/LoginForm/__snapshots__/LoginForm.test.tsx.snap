// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form renders correctly 1`] = `
.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: 100%;
}

.c3 {
  border: 1px solid #909AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c3:disabled,
.c3:hover,
.c3:active,
.c3:focus {
  outline: none;
}

.c2 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #EF6517;
  border: 1px solid black;
  border-radius: 0.5rem;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c5:disabled,
.c5:hover,
.c5:active,
.c5:focus {
  outline: none;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  width: 25.625rem;
}

.c6 {
  color: #EF6517;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <h2
    style={
      {
        "color": "#FCFEFF",
        "fontFamily": "Poppins, sans-serif",
        "fontSize": "2rem",
        "fontWeight": 600,
        "letterSpacing": "2px",
        "lineHeight": 1.2,
        "textAlign": "center",
      }
    }
  >
    Login
  </h2>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="login"
      size="medium"
    >
      Login
    </label>
    <input
      className="c3"
      id="login"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter your login"
      type="text"
    />
  </div>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="password"
      size="medium"
    >
      Password
    </label>
    <input
      className="c3"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter your password"
      type="password"
    />
  </div>
  <span
    className="c4"
  />
  <button
    className="c5"
    type="submit"
  >
    Login
  </button>
  <a
    className="c6"
    href="/resetPassword"
    onClick={[Function]}
    palette={
      {
        "accent": "#0E172D",
        "background": "#0A1122",
        "backgroundGrey": "#F5F5F5",
        "badge": "#CF0000",
        "danger": "#FF4438",
        "default": "#000",
        "footer": "#3F4476",
        "greyDark": "#767676",
        "greyLight": "#F5F6F8",
        "inputGrey": "#4B4B4B",
        "light": "#FCFEFF",
        "primaryDark": "#4553DD",
        "primaryLight": "#909AFF",
        "secondaryDark": "#EF6517",
        "secondaryLight": "#FF7C33",
        "shadow": "#4B4D62",
        "sidebarTabs": "#CACACA",
        "success": "#83E57A",
        "successDeep": "#3A9832",
        "text": "#1B1A1A",
        "widgetBackground": "#fff",
      }
    }
  >
    Forgot your password?
  </a>
</form>
`;
